primitives/src/assets.rs

```diff
@@ -444,7 +448,7 @@ where
         amount: Balance,
     ) -> Result<(), FungibleLedgerError> {
         Self::ensure_valid(asset_id)?;
-        Self::can_deposit(asset_id, beneficiary, amount)?;
+        Self::can_deposit(asset_id, beneficiary, amount, true)?;
```

Comment:
I think you can just move the can_deposit check in the if branches.
For native assets use mint = false
For non-native assets use mint = true

Reply:
Also all the manta-pay calls should use mint = false.

Reply:
Also the `can_deposit` integratikon tests for native asset should use mint = false.